### PR TITLE
Table of categories and filter tindercards by category

### PR DESCRIPTION
**Pull request by BboyStatix**

### CODE REVIEW

File: app/src/main/java/com/example/sagar/popupshops_buyerside/Shop/add.java [hunk 1]
```diff
@@ -111,11 +135,27 @@ public void onFailure(@NonNull Exception e) {
...
                                                         @Override
                                                         public void onSuccess(UploadTask.TaskSnapshot taskSnapshot) {
-                                                            dbRef.setValue(
+                                                            itemRef.setValue(
                                                                     new Item(
                                                                             categoryString, Float.parseFloat(priceString), description, taskSnapshot.getMetadata().getDownloadUrl().toString(), Integer.parseInt(stockString)
                                                                     )
                                                             );
+
+                                                            categoryRef.runTransaction(new Transaction.Handler() {
```
Comment (BboyStatix) [inline app/src/main/java/com/example/sagar/popupshops_buyerside/Shop/add.java]: If category already exists, then dont add category to the generalized category table

File: app/src/main/java/com/example/sagar/popupshops_buyerside/Shop/add.java [hunk 2]
```diff
@@ -111,11 +135,27 @@ public void onFailure(@NonNull Exception e) {
...
+                                                                    };
+                                                                    return null;
+                                                                }
+
+                                                                @Override
+                                                                public void onComplete(DatabaseError databaseError, boolean b, DataSnapshot dataSnapshot) {
+
+                                                                }
+                                                            });
                                                             //add shop id to item record
```
Comment (BboyStatix) [inline app/src/main/java/com/example/sagar/popupshops_buyerside/Shop/add.java:159]: This part is basically to add categories to the general category table. If it already exists then dont add.

File: app/src/main/java/com/example/sagar/popupshops_buyerside/Shop/add.java [hunk 3]
```diff
@@ -90,18 +97,35 @@ public void onClick(View view) {
...
+                String[] categoryValues = categoryHashMap.values().toArray(new String[categoryHashMap.size()]);
+                final ArrayAdapter<String> adapter = new ArrayAdapter<String>(getApplicationContext(), android.R.layout.simple_dropdown_item_1line, categoryValues);
+                categoryInput.setAdapter(adapter);
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+                System.out.println("The read failed: " + databaseError.getCode());
+            }
+        });
```
Comment (BboyStatix) [inline app/src/main/java/com/example/sagar/popupshops_buyerside/Shop/add.java:118]: This part is for the autocompletetextview in the buyer side inputting category. But for some reason its not working.

File: app/src/main/java/com/example/sagar/popupshops_buyerside/MainActivity.java [hunk 4]
```diff
@@ -126,7 +172,12 @@ public void onKeyEntered(String key, GeoLocation location) {
...
                         Log.d(TAG, "onChildAdded:" + dataSnapshot.getKey());
                         Item item = dataSnapshot.getValue(Item.class);
-                        mSwipeView.addView(new ItemCard(mContext, item, mSwipeView));
+                        final Spinner dropdown = (Spinner) findViewById(R.id.spinner1);
+                        items.add(item);
+                        String selectedItem = dropdown.getSelectedItem().toString();
+                        if(selectedItem == "All" || (item.getItemCategory() != null && item.getItemCategory().equals(selectedItem))){
+                            mSwipeView.addView(new ItemCard(mContext, item, mSwipeView));
+                        }
                     }
```
Comment (BboyStatix) [inline app/src/main/java/com/example/sagar/popupshops_buyerside/MainActivity.java:181]: This file is to do with the main buyer activity. But the layout rules are weird so cant select dropdown

File: app/src/main/java/com/example/sagar/popupshops_buyerside/Shop/add.java [hunk 5]
```diff
@@ -90,18 +97,35 @@ public void onClick(View view) {
...
-        categoryInput.setAdapter(suggest(this));
+        ArrayAdapter<String> adapter = new ArrayAdapter<String>(this, android.R.layout.simple_dropdown_item_1line, new String[]{});
+        categoryInput.setAdapter(adapter);
+        categoryInput.setThreshold(1);
+
+        categoryRef.addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                HashMap<String, String> categoryHashMap = (HashMap<String, String>) dataSnapshot.getValue();
+                String[] categoryValues = categoryHashMap.values().toArray(new String[categoryHashMap.size()]);
```
Comment (sacrarat) [inline app/src/main/java/com/example/sagar/popupshops_buyerside/Shop/add.java:108]: @BboyStatix sure the thing inside the `toArray()` shouldnt just be categoryHashMap.size()?

**Pull request merged**